### PR TITLE
docs: Add Jenkins CI doc. Fixes #2067

### LINKED ISSUE #2067: Running dagger on Jenkins doc
### What is the issue?

We need an example of how to run dagger within a Jenkins pipeline. Will use a jenkinsfile approach.

### PR DESCRIPTION
**Pull request by jpadams**
Signed-off-by: Jeremy Adams <jeremy@dagger.io>

### CODE REVIEW

**Comment by jpadams** [comment]
Not sure if Lint is essential. I notice other docs fail `markdownlint` as well.
image

**Review by shykes — APPROVED** [review]
For the sake of getting a first version out there, let's merge as is.

I do agree that we shoud remove that `dagger project init` and `dagger project update`.

* I understand why some users are doing it
    * it's partly because we don't explain the best practices clearly...
    * and partly because there is some pain/inconvenience in our current implementation that creates the temptation of deviating from our best practices (eg. we vendor all cue.mod in bulk...)
* However, it's one thing for our users to deviate from best practices, it's another for us to do it in our own examples. I'd rather we 1) stick to the best practices, 2) go out of our way to help users work around faulty implementation, 3) fix our implementation ASAP so that there's no more reason to deviate.

--> All of which we can do as a follow-up PR :)

File: docs/getting-started/1201-ci-environment.md [hunk 1]
```diff
@@ -105,7 +105,38 @@ build:
...
+        //  dagger project init
+        //  dagger project update
+        //'''
+      }
+    }
+    stage("do") {
+      steps {
+        //substitute your action name for 'helloworld'
+        //e.g. 'build' or 'push' or whatever you've created!
+        sh 'dagger do helloworld --log-format=plain'
```
Comment (jglick) [inline docs/getting-started/1201-ci-environment.md:134]: Could probably use `tty` with https://plugins.jenkins.io/ansicolor/

**Review by jglick — COMMENTED** [review]
All pretty basic. For example, if you run tests, you would have to dig through the build log to see failure details. https://docs.dagger.io/1226/handling-outputs#writing-outputs-to-the-filesystem implies you could write out the likes of `TEST-*.xml` to the Jenkins agent workspace for the `junit` step to pick up; or write out non-image outputs for `archiveArtifacts`?

**Pull request merged**